### PR TITLE
Add PR queue action to queue PRs

### PR DESCRIPTION
**Pull request by xurtis**
This adds an action that manages a queue of PRs to be merged via BitBucket by commenting on the PRs to prompt action.

### CODE REVIEW

**Comment by xurtis** [comment]
@lsf37 I'd like to get this on all repositories that currently still need to go via BitBucket to get merged to automate the management of ordering PRs and getting them all merged. Would be good to test it on a less-critical repository first though

File: pr-queue/README.md
Comment (lsf37) [inline pr-queue/README.md:47]:
```suggestion
chosen.
```

File: pr-queue/README.md
Comment (lsf37) [inline pr-queue/README.md:58]: please update example name

File: pr-queue/README.md
Comment (lsf37) [inline pr-queue/README.md:55]: please update example name

File: pr-queue/action.yml
Comment (lsf37) [inline pr-queue/action.yml:6]:
```suggestion
description: 'Manages queuing of PRs via Bitbucket at D61'
```

File: pr-queue/index.js
Comment (lsf37) [inline pr-queue/index.js:40]: This should work, but there is also `github.repository_owner` which should be passed as context to the action. Not sure if it's worth changing, just pointing out that it exists.

File: pr-queue/index.js
Comment (lsf37) [inline pr-queue/index.js:69]: We'd usually want 2 approvals. (1 is possible, but should be an exception, e.g. if the PR is from someone who already has commit rights).

File: pr-queue/index.js
Comment (lsf37) [inline pr-queue/index.js:82]: This will return true if there is even one test passing. I think we want this to be all tests, i.e. something like `checks.filter(is_success).length == checks.length`

**Review by lsf37 — APPROVED** [review]
I like the idea and am happy to try it out. Only minor comments to fix up.

**Comment by lsf37** [comment]
The link checker is failing on `https://sel4.systems/Foundation/About/` which is now `https://sel4.systems/Foundation/`. Would be nicer in a separate PR, but I don't want to change the repo settings atm. Are you Ok adding a small commit that updates this link in `CONTRIBUTING.md`?

File: pr-queue/index.js
Comment (xurtis) [reply pr-queue/index.js:69]: This is more a prompt to get any reviewers at all, the actual number of reviewers should be checked by the assignee who then goes to merge.

Comment (lsf37) [reply pr-queue/index.js:69]: Wouldn't we want to prefer a PR that already has all the necessary reviewers? I guess one approval should still beat no approvals, though. Happy to see first if it makes any difference in practice if you want.

**Pull request merged**
